### PR TITLE
Feat/kbdlight

### PR DESCRIPTION
**Pull request by aymanbagabas**
Merge feat/kbdlight into master

### CODE REVIEW

**Comment by sermart1234** [comment]
Ok

File: huawei-wmi.c [hunk 1]
```diff
@@ -600,6 +873,567 @@ static void huawei_wmi_fn_lock_exit(struct device *dev)
...
+
+
+
+/*FN led (read only)*/
+
+static void huawei_wmi_led_fake_set(struct led_classdev *led_cdev,
+		enum led_brightness brightness)
+{
+	printk(KERN_INFO "huawei_wmi: %s read only\n", led_cdev->name);
+}
```
Comment (aymanbagabas) [inline huawei-wmi.c:1381]: Why do we need this?

File: huawei-wmi.c [hunk 2]
```diff
@@ -600,6 +873,567 @@ static void huawei_wmi_fn_lock_exit(struct device *dev)
...
+
+static enum led_brightness huawei_wmi_fn_led_get(struct led_classdev *led_cdev)
+{
+	u64 value;
+	int err;
+
+	err = read_ec_by_name("FNKL", &value);
```
Comment (aymanbagabas) [inline huawei-wmi.c:1388]: Can't we use the HWMI interface for this? It feels hacky

File: huawei-wmi.c [hunk 3]
```diff
@@ -600,6 +873,567 @@ static void huawei_wmi_fn_lock_exit(struct device *dev)
...
+}
+
+/*Keyboard backlight level (read only)*/
+
+static enum led_brightness huawei_wmi_KCMS_get(struct led_classdev *led_cdev)
+{
+	u64 value;
+	int err;
+
+	err = read_ec_by_name("KCMS", &value);
```
Comment (aymanbagabas) [inline huawei-wmi.c:1416]: Same, can we use HWMI?

File: huawei-wmi.c [hunk 4]
```diff
@@ -808,6 +1650,24 @@ static int huawei_wmi_probe(struct platform_device *pdev)
 	if (wmi_has_guid(HWMI_METHOD_GUID)) {
 		mutex_init(&huawei_wmi->wmi_lock);
 
+		huawei_wmi_KCMS_setup(&pdev->dev);
+		huawei_wmi_fn_led_setup(&pdev->dev);
```
Comment (aymanbagabas) [inline huawei-wmi.c:1654]: These should be moved into `huawei_wmi_leds_setup`

**Review by aymanbagabas — COMMENTED** [review]
@sermart1234 few comments before merging this

**Comment by aymanbagabas** [comment]
@sermart1234 any updates on this?

**Comment by sermart1234** [comment]
It is OK

**Pull request merged**